src/ngResource/resource.js

```diff
@@ -586,9 +590,11 @@ angular.module('ngResource', ['ng']).
                          'requests, you should use the `cancellable` option.');
               delete action.timeout;
             }
-            cancellable = angular.isDefined(action.cancellable) ? action.cancellable :
-                         (options && angular.isDefined(options.cancellable)) ? options.cancellable :
-                         provider.defaults.cancellable;
+          } else {
+            if(cancellable) {
+              simulatedTimeout = action.timeout;
+              delete action.timeout;
```

Comment:
I don't think it is overwritten. `action.timeout` is not the same as `httpConfig.timeout`.
But we shouldn't delete it if we don't have to :smiley: